code/espurna/config/prototypes.h

```diff
@@ -128,7 +129,33 @@ void i2c_read_buffer(uint8_t address, uint8_t * buffer, size_t len);
 // -----------------------------------------------------------------------------
 // OTA
 // -----------------------------------------------------------------------------
-#include "ESPAsyncTCP.h"
+
+#if OTA_ARDUINOOTA_SUPPORT
+    #include <ArduinoOTA.h>
+#endif
+
+#if OTA_CLIENT == OTA_CLIENT_ASYNCTCP
+    #include <ESPAsyncTCP.h>
+#endif
+
+#if OTA_CLIENT == OTA_CLIENT_HTTPUPDATE
+    #include <ESP8266HTTPClient.h>
+    #include <ESP8266httpUpdate.h>
+#endif
+
+#if SSL_CLIENT != SSL_CLIENT_NONE
+
+    #include <WiFiClientSecure.h>
+
+    // TODO: add DigiCert CA for github?
+    #if SSL_CLIENT_BEARSSL && OTA_SSL_CLIENT_INCLUDE_CA
+    #include "static/ota_ssl_client_ca.h"
+    #else
```

Comment:
``#else if SSL_CLIENT_BEARSSL``?

Reply:
I think I wanted to add something here using .cpp files and weaken the progmem char symbol, but that did not quite work. This probably can even be just a check for include_ca flag, progmem'ed array will be discarded anyways as unused variable.